flutter/lib/src/user_interaction/sentry_user_interaction_widget.dart

```diff
@@ -0,0 +1,335 @@
+      transactionNameSource: SentryTransactionNameSource.component,
+    );
+
+    final activeTransaction = _activeTransaction;
+    if (activeTransaction != null) {
+      if (_lastTappedWidget?.element.widget == element.widget &&
+          _lastTappedWidget?.eventType == tappedWidget.eventType &&
+          !activeTransaction.finished) {
+        // ignore: invalid_use_of_internal_member
+        activeTransaction.scheduleFinish();
+        return;
+      } else {
+        activeTransaction.finish();
+        _hub.configureScope((scope) {
+          if (scope.span == activeTransaction) {
+            scope.span = null;
+          }
+        });
+        _activeTransaction = null;
+        _lastTappedWidget = null;
+      }
+    }
+
+    _lastTappedWidget = tappedWidget;
+
+    bool hasRunningTransaction = false;
+    _hub.configureScope((scope) {
+      if (scope.span != null) {
+        hasRunningTransaction = true;
+      }
+    });
+
+    if (hasRunningTransaction) {
+      return;
+    }
+
+    // TODO: mobile vitals
+    _activeTransaction = _hub.startTransactionWithContext(
+      transactionContext,
+      waitForChildren: true,
+      autoFinishAfter:
+          // ignore: invalid_use_of_internal_member
+          _options?.idleTimeout,
+      trimEnd: true,
+    );
+
+    // if _enableAutoTransactions is enabled but there's no traces sample rate
+    if (_activeTransaction is NoOpSentrySpan) {
+      return;
+    }
+
+    _hub.configureScope((scope) {
+      scope.span ??= _activeTransaction;
+    });
+  }
+
+  String _findDescriptionOf(Element element, bool allowText) {
+    var description = '';
+
+    // traverse tree to find a suiting element
+    void descriptionFinder(Element element) {
+      bool foundDescription = false;
+
+      final widget = element.widget;
+      if (allowText && widget is Text) {
+        final data = widget.data;
+        if (data != null && data.isNotEmpty) {
+          description = data;
+          foundDescription = true;
+        }
+      } else if (widget is Semantics) {
+        if (widget.properties.label?.isNotEmpty ?? false) {
+          description = widget.properties.label!;
+          foundDescription = true;
+        }
+      } else if (widget is Icon) {
+        if (widget.semanticLabel?.isNotEmpty ?? false) {
+          description = widget.semanticLabel!;
+          foundDescription = true;
+        }
+      }
+
+      if (!foundDescription) {
+        element.visitChildren(descriptionFinder);
+      }
+    }
+
+    element.visitChildren(descriptionFinder);
+
+    return description;
+  }
+
+  UserInteractionWidget? _getElementAt(Offset position) {
+    // WidgetsBinding.instance.renderViewElement does not work, so using
+    // the element from createElement
+    final rootElement = _clickTrackerElement;
+    if (rootElement == null || rootElement.widget != widget) {
+      return null;
+    }
+
+    UserInteractionWidget? tappedWidget;
+
+    void elementFinder(Element element) {
```

Comment:
h: Consider this:

```dart
Card(
                child: GestureDetector(
                  onTap: () => {
                    //open Card
                  },
                  child: Stack(
                    children: [
                      //fancy card layout
                      ElevatedButton(
                        onPressed: () => {
                          //mark card as favorite
                        },
                        child: const Text('Favorite'),
                      ),
                    ],
                  ),
                ),
              )
```

If a user press the "Favorite" `ElevatedButton`, the element found will always be the `GestureDetector`.
I guess this will be a problem to really detect a potential problem.

Reply:
It's kind ok to keep this as it is. Since UI Transactions are meant to a bag for spans. Maybe we're ok to not have the perfect origin of a transaction as long we have a transaction.

But if you can figure it out how to solve this would be nice.

Reply:
Very good catch, the `elementFinder` always goes to the `GestureDetector`.
@ueman any ideas what to do here? since we don't know the instance of the `onTap` or `onPressed` to compare, I don't know if we can solve it.
@brustolin  The question is, does this even make sense? Having a parent `onTap` and an `onPressed` child? Is this a real use case? Flutter always executes the `onPressed` from the `ElevatedButton`, always ignores the `onTap` from `GestureDetector`, maybe we should just go till the very end of the tree and use the last found child?

Reply:
Yes, this is a real use case.
I had used this a lot.

Reply:
> @ueman any ideas what to do here? since we don't know the instance of the onTap or onPressed to compare, I don't know if we can solve it.

> maybe we should just go till the very end of the tree and use the last found child?

Yeah, maybe don't stop searching on the first hit, but for the smallest hit?

Maybe there's a way to get some information from the gesture arena (which is the mechanism which decides which widget was tapped)? There's some information here https://docs.flutter.dev/development/ui/advanced/gestures#gesture-disambiguation

Reply:
It turns out that going to the latest level of `child` is also wrong, because it ends up with an `InkWell` which is a child of the desired `ElevatedButton`.
Will do a bit of research.

Reply:
For now, I dropped support for the nested `GestureDetector` so the PR is unblocked, I will keep searching on how to apply the gesture disambiguation
https://docs.flutter.dev/development/ui/advanced/gestures#gesture-disambiguation